packages/sync/src/offline/OfflineRestoreHandler.ts

```diff
@@ -42,27 +42,26 @@ export class OfflineRestoreHandler {
     if (!this.hasOfflineData()) { return; }
 
     logger("Replying offline mutations after application restart");
-    // return as promise, but in the end clear the storage
-    const uncommittedOfflineMutation: OperationQueueEntry[] = [];
 
-    await Promise.all(this.offlineData.map(async (item) => {
-      try {
-        await this.apolloClient.mutate({
-          variables: item.operation.variables,
-          mutation: item.operation.query,
-          context: item.operation.getContext
-        });
-      } catch (e) {
-        // set the errored mutation to the stash
-        uncommittedOfflineMutation.push(item);
-      }
-    }));
+    // clear offline data from storage and push all offline mutations
+    // back to offline queue
 
-    // wait before it was cleared
-    await this.clearOfflineData();
+    this.clearOfflineData();
 
-    // then add again the uncommited storage
-    this.addOfflineData(uncommittedOfflineMutation);
+    this.offlineData.forEach((item, index) => {
+      const last = index === this.offlineData.length - 1;
```

Comment:
Why we need this?

Reply:
Idea is good but we missing documentation. I'm curious how we can document all those extra __ that are added now. It will be hard to get into this after month etc. This definitely works, but what we should be also look for is to have this maintainable.